core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -102,16 +102,12 @@
     public static final ParseField PROFILE_FIELD = new ParseField("profile");
     public static final ParseField SEARCH_AFTER = new ParseField("search_after");
 
-    private static final SearchSourceBuilder PROTOTYPE = new SearchSourceBuilder();
-
-    public static SearchSourceBuilder readSearchSourceFrom(StreamInput in) throws IOException {
-        return PROTOTYPE.readFrom(in);
-    }
-
     public static SearchSourceBuilder parseSearchSource(XContentParser parser, QueryParseContext context,
                                                         AggregatorParsers aggParsers, Suggesters suggesters)
             throws IOException {
```

Comment:
nit: I wonder if we should rename this to "fromXContent" like we do in most of the queries, suggesters etc... No strong opinion here though.